src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyNames.cs

```diff
@@ -0,0 +1,194 @@
+                                                                             isDataflowRule: false);
+        internal static DiagnosticDescriptor FieldRule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessageField,
+                                                                             DiagnosticCategory.Design,
+                                                                             RuleLevel.BuildWarning,
+                                                                             description: s_localizableDescription,
+                                                                             isPortedFxCopRule: false,
+                                                                             isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(PropertyRule, FieldRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(
+                (compilationStartContext) =>
+                {
+                    INamedTypeSymbol? jsonConstructorAttributeNamedSymbol = compilationStartContext.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextJsonSerializationJsonConstructorAttribute);
+                    if (jsonConstructorAttributeNamedSymbol == null)
+                    {
+                        return;
+                    }
+
+                    var paramAnalyzer = new ParameterAnalyzer(jsonConstructorAttributeNamedSymbol);
+
+                    compilationStartContext.RegisterOperationBlockStartAction(
+                        (startOperationBlockContext) =>
+                        {
+                            if (startOperationBlockContext.OwningSymbol is IMethodSymbol method
+                                && !paramAnalyzer.ShouldAnalyzeMethod(method))
+                            {
+                                return;
+                            }
+
+                            startOperationBlockContext.RegisterOperationAction(
+                                context => ParameterAnalyzer.AnalyzeOperationAndReport(context),
+                                OperationKind.ParameterReference);
+                        });
+                });
+        }
+
+        private sealed class ParameterAnalyzer
+        {
+            private readonly INamedTypeSymbol _jsonConstructorAttributeInfoType;
+
+            public ParameterAnalyzer(INamedTypeSymbol jsonConstructorAttributeInfoType)
+            {
+                _jsonConstructorAttributeInfoType = jsonConstructorAttributeInfoType;
+            }
```

Comment:
Only one method is using the field. I personally prefer this being as a static class with jsonConstructorAttributeType being passed to the only method needs it.